brvs-core/src/main/java/iroha/validation/transactions/provider/impl/BasicTransactionProvider.java

```diff
@@ -102,17 +104,22 @@ public BasicTransactionProvider(
   }
 
   private void monitorIrohaPending() {
-    irohaReliableChainListener
-        .getAllPendingTransactions(registrationProvider.getRegisteredAccounts())
-        .forEach(transactionBatch -> {
-              // if only BRVS signatory remains
-              if (isBatchSignedByUsers(transactionBatch)) {
-                if (savedMissingInStorage(transactionBatch)) {
-                  cacheProvider.put(transactionBatch);
+    try {
+      irohaReliableChainListener
+          .getAllPendingTransactions(registrationProvider.getRegisteredAccounts())
+          .forEach(transactionBatch -> {
+                // if only BRVS signatory remains
+                if (isBatchSignedByUsers(transactionBatch)) {
```

Comment:
I would rewrite it like that `if (isBatchSignedByUsers(transactionBatch) && savedMissingInStorage(transactionBatch))`

Reply:
ok